File: examples/dotnet/getting-started/pulumi-and-structurizr/Systems/PaymentProvider/PaymentProvider.cs
using Structurizr;

internal class PaymentProvider
{
    public SoftwareSystem System { get; private set; }

    public PaymentProvider(Workspace workspace, Person customer)
    {
        System = workspace.Model.AddSoftwareSystem(Location.External,  "Payment Provider", "Online payment provider");
        customer.Uses(System, "pay for stuff");
    }
}

File: labs/nodejs/pulumi-and-structurizr/systems/paymentProvider/paymentProvider.ts
import { Workspace, Location, SoftwareSystem, Person } from 'structurizr-typescript';

export class PaymentProvider {
    public system: SoftwareSystem;

    constructor(workspace: Workspace, customer: Person) {
        this.system = workspace.model.addSoftwareSystem('Payment Provider', 'Online payment provider')!;
        this.system.location = Location.External;
        
        customer.uses(this.system, 'pay for stuff');
    }
};